registry/storage/driver/hdfsweb/hdfs_test.go

```diff
@@ -0,0 +1,147 @@
+package hdfsweb
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"testing"
+
+	log "github.com/Sirupsen/logrus"
+	ctx "github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/testsuites"
+	"gopkg.in/check.v1"
+)
+
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { check.TestingT(t) }
+
+var HdfsDriverConstructor func(rootDirectory string) (*Driver, error)
+var skipHdfs func() string
+
+func init() {
+	nameNodeHost := os.Getenv("HDFS_NAME_NODE_HOST")
+	nameNodePort := os.Getenv("HDFS_NAME_NODE_PORT")
+	root, err := ioutil.TempDir("", "driver-")
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(root)
+	HdfsDriverConstructor = func(rootDirectory string) (*Driver, error) {
+		var blockSize, bufferSize, replication int
+		var err error
+		if nameNodeHost == "" {
+			return nil, fmt.Errorf("not provide HDFS_NAME_NODE_HOST")
+		}
+
+		if nameNodePort == "" {
+			return nil, fmt.Errorf("not provide HDFS_NAME_NODE_HOST")
+		}
+
+		if v := os.Getenv("HDFS_BLOCK_SIZE"); v != "" {
+			blockSize, err = strconv.Atoi(v)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			blockSize = defaultBlockSize
+		}
+
+		if v := os.Getenv("HDFS_BUFFER_SIZE"); v != "" {
+			bufferSize, err = strconv.Atoi(v)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			bufferSize = defaultBufferSize
+		}
+
+		if v := os.Getenv("HDFS_REPLICATION"); v != "" {
+			replication, err = strconv.Atoi(v)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			replication = defaultReplication
+		}
+
+		return New(DriverParameters{
+			nameNodeHost,
+			nameNodePort,
+			rootDirectory,
+			blockSize,
+			bufferSize,
+			replication,
+		})
+	}
+
+	// Skip HDFS storage driver tests if environment variable parameters are not provided
+	skipHdfs = func() string {
+		if nameNodeHost == "" || nameNodePort == "" {
+			return "Must set HDFS_NAME_NODE_HOST, HDFS_NAME_NODE_PORT and HDFS_ROOT_DIRECTORY to run HDFS tests"
+		}
+		return ""
+	}
+
+	testsuites.RegisterSuite(func() (storagedriver.StorageDriver, error) {
+		return HdfsDriverConstructor(root)
+	}, skipHdfs)
+}
+
+func TestEmptyRootList(t *testing.T) {
+	if skipHdfs() != "" {
+		t.Skip(skipHdfs())
+	}
+
+	validRoot, err := ioutil.TempDir("", "driver-")
+	if err != nil {
+		log.Errorf("unexpected error creating temporary directory: %v", err)
```

Comment:
These log lines shouldn't be necessary inside tests. I think `t.Fatalf` is sufficient.